src/test/java/uk/ac/ebi/embl/api/validation/fixer/entry/NonAsciiCharacterFixTest.java

```diff
@@ -0,0 +1,91 @@
+package uk.ac.ebi.embl.api.validation.fixer.entry;
+
+import org.junit.Before;
+import org.junit.Test;
+import uk.ac.ebi.embl.api.entry.Entry;
+import uk.ac.ebi.embl.api.entry.EntryFactory;
+import uk.ac.ebi.embl.api.entry.Text;
+import uk.ac.ebi.embl.api.entry.feature.Feature;
+import uk.ac.ebi.embl.api.entry.feature.FeatureFactory;
+import uk.ac.ebi.embl.api.entry.qualifier.Qualifier;
+import uk.ac.ebi.embl.api.entry.qualifier.QualifierFactory;
+import uk.ac.ebi.embl.api.entry.reference.*;
+import uk.ac.ebi.embl.api.validation.Severity;
+import uk.ac.ebi.embl.api.validation.ValidationMessageManager;
+import uk.ac.ebi.embl.api.validation.ValidationResult;
+import uk.ac.ebi.embl.api.validation.plan.EmblEntryValidationPlanProperty;
+
+import java.sql.SQLException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class NonAsciiCharacterFixTest {
+    private Entry entry;
+    private NonAsciiCharacterFix fixer;
+    private Reference reference1;
+    private Person person1;
+
+    @Before
+    public void setUp() throws SQLException
+    {
+        ValidationMessageManager.addBundle(ValidationMessageManager.STANDARD_FIXER_BUNDLE);
+        EntryFactory entryFactory = new EntryFactory();
+        ReferenceFactory referenceFactory = new ReferenceFactory();
+        EmblEntryValidationPlanProperty property=new EmblEntryValidationPlanProperty();
+        reference1 = referenceFactory.createReference();
+        Submission submission = (new ReferenceFactory()).createSubmission(referenceFactory.createPublication());
+        reference1.setPublication(submission);
+        person1 = referenceFactory.createPerson();
+        entry = entryFactory.createEntry();
+        fixer = new NonAsciiCharacterFix();
+        fixer.setEmblEntryValidationPlanProperty(property);
+    }
+
+    @Test
+    public void testCheck_valid()
+    {
+        ValidationResult result = fixer.check(entry);
+        assertTrue(result.isValid());
+    }
+
+    @Test
+    public void testCheck_fixedComment() {
+        entry.setComment(new Text("non-ascii cömment"));
+        ValidationResult result = fixer.check(entry);
+        assertEquals(1, result.count("AsciiCharacterFix_1", Severity.FIX));
+        assertEquals(entry.getComment(), "non-ascii comment"); // ö -> o
+    }
+
+    @Test
+    public void testCheck_knownNotFixedChar()
+    {
+        entry.setComment(new Text("non-ascii c�mment")); // expected: not caught and not possible to fix
+        ValidationResult result = fixer.check(entry);
```

Comment:
May be good to compare what has been set after fix, or to assert number of fix is 0 , like in the above tests.

 assertEquals(1, result.count("AsciiCharacterFix_1", Severity.FIX));
        assertEquals(entry.getComment(), "non-ascii comment");

Reply:
@snathanvj this is one scenario which we can't fix and it will fallback to the AsciiCharacterCheck to catch and reject.

Reply:
added
```
assertEquals(0, result.count(Severity.FIX));
```
to be clear we can't fix this type of char as we don't know what to map it to.